diff --git a/extras/Build/CMake/JUCEUtils.cmake b/extras/Build/CMake/JUCEUtils.cmake
index 8a6b0d6..415256d 100644
--- a/extras/Build/CMake/JUCEUtils.cmake
+++ b/extras/Build/CMake/JUCEUtils.cmake
@@ -1231,14 +1231,14 @@ function(_juce_set_plugin_target_properties shared_code_target kind)
         _juce_adhoc_sign(${target_name})
 
         get_target_property(vst3_auto_manifest ${shared_code_target} JUCE_VST3_AUTO_MANIFEST)
-
-        add_custom_command(TARGET ${target_name} POST_BUILD
-            COMMAND ${CMAKE_COMMAND} -E echo "removing moduleinfo.json"
-            COMMAND ${CMAKE_COMMAND} -E remove -f
-                "${output_path}/Contents/moduleinfo.json"
-                "${output_path}/Contents/Resources/moduleinfo.json")
-
         if(vst3_auto_manifest)
+            add_custom_command(TARGET ${target_name} POST_BUILD
+                COMMAND ${CMAKE_COMMAND} -E echo "removing moduleinfo.json"
+                COMMAND ${CMAKE_COMMAND} -E remove -f
+                    "${output_path}/Contents/moduleinfo.json"
+                    "${output_path}/Contents/Resources/moduleinfo.json"
+                VERBATIM)
+
             juce_enable_vst3_manifest_step(${shared_code_target})
         endif()
     elseif(kind STREQUAL "VST")
